test/StaticValidationTests/Program.cs

```diff
@@ -0,0 +1,297 @@
+                        ValidationTestCommon.LogLine($"Experimental Attribute found in {typeDefNameSpace}.{typeDefName}");
+                        found = true;
+                    }
+                }
+            }
+            return found;
+        }
+
+        private static bool IsAttributeOfType(MetadataReader metadataReader, CustomAttribute customAttribute, string attribute)
+        {
+            switch (customAttribute.Constructor.Kind)
+            {
+                case HandleKind.MemberReference:
+                    MemberReference memberReference = metadataReader.GetMemberReference((MemberReferenceHandle)customAttribute.Constructor);
+                    TypeReference typeRef = metadataReader.GetTypeReference((TypeReferenceHandle)memberReference.Parent);
+
+                    string typeRefNameSpace = metadataReader.GetString(typeRef.Namespace);
+                    string typeRefName = metadataReader.GetString(typeRef.Name);
+                    string typeRefType = $"{typeRefNameSpace}.{typeRefName}";
+                    if (typeRefType.Equals(attribute))
+                    {
+                        return true;
+                    }
+                    return false;
+
+                default:
+                    return false;
+            }
+        }
+
+        private static bool OutputContainsLine(CommandLineProcess cmd, string line)
+        {
+            foreach (string s in cmd.StandardOut)
+            {
+                if (s.Trim().Contains(line))
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        private static bool CheckDigitalSignature(string signtool, string file)
+        {
+            CertValidation extraValidation = CertValidations.Where(c => string.Equals(c.File, Path.GetFileName(file), StringComparison.OrdinalIgnoreCase)).FirstOrDefault();
+
+            string extraOptions = extraValidation?.ExtraSignToolFlags ?? "";
+            string arguments = $"verify /pa /v {extraOptions} /r \"{RootCertificate}\" \"{file}\"";
+            bool hasValidDigitalSignature = false;
+
+            try
+            {
+                var cmd = new CommandLineProcess(signtool, arguments);
+                hasValidDigitalSignature = cmd.Run();
+                if (!hasValidDigitalSignature)
+                {
+                    ValidationTestCommon.LogLine("ERROR: SignTool verify returned error.");
+                    ValidationTestCommon.LogLine($"SignTool command line: signtool {arguments}");
+                    ValidationTestCommon.LogLine("SignTool output:");
+                    ValidationTestCommon.LogLine("=====================");
+                    ValidationTestCommon.LogLine(string.Join("\r\n", cmd.StandardOut));
+                    ValidationTestCommon.LogLine("=====================");
+                }
+
+                if (extraValidation != null)
```

Comment:
Is this block applicable even when the digital signature is invalid?

Reply:
So this is a vestigial from the Aggregator code where it had to check for different signatures for Microsoft.Internal.FrameworkUdk.dll but we don't need that in Foundation hence CertValidations in line 63 is empty.

But I am keeping this logic just incase.